src/v/kafka/server/handlers/fetch.cc

```diff
@@ -1702,7 +1706,8 @@ ss::future<response_ptr> op_context::send_response() && {
         final_response.data.topics.back().partitions.push_back(std::move(r));
     }
 
-    return rctx.respond(std::move(final_response));
+    return rctx.respond(std::move(final_response))
+      .finally([units = std::move(fetch_memory_units)] {});
```

Comment:
So this is where the units will finally be released, right?

It sort feels like these should be released in exactly the same place as the units from every other request type, since that's the one place we decided on and we don't have to try to analyze two separate locations.